schema/gen/go/genUnion.go

```diff
@@ -478,9 +479,9 @@ func (g unionBuilderGenerator) emitMapAssemblerMethods(w io.Writer) {
 				return ma.ca{{ add $i 1 }}, nil
 				{{- end}}
 			{{- end}}
-			default:
-				return nil, ipld.ErrInvalidKey{TypeName:"{{ .PkgName }}.{{ .Type.Name }}", Key:&_String{k}}
+			{{- end}}
 			}
+			return nil, ipld.ErrInvalidKey{TypeName:"{{ .PkgName }}.{{ .Type.Name }}", Key:&_String{k}}
```

Comment:
Does `go vet` prefer this?

I tend to like writing a default clause with a return statement in it, because then it makes sure I don't forget a return statement in every other branch in every other part of the switch (if there's no return outside the switch, then the compiler would yell at me if I forget one in any of the cases).

Reply:
I thought you might flag this :P The vet rule at play here was just the dead code one, because in the zero-case scenario, we would produce something like:

```
switch {
default:
    return someError
}
return nil // dead!
```

So the simplest fix, from the POV of the generator, was to flip the returns around so that this could never happen. But I can just add some extra template logic to instead emit different code when there are zero cases. Would you prefer that?

Regarding the safety of the code I agree, but I think it matters less here since it's generated code. But I'm happy to undo a change that's not really necessary to appease vet.

Reply:
Except we didn't generate that trailing `return nil`, either, so... the vet dead code detector on the zero-case scenario just doesn't like a lonely default?  (I guess that's not unreasonable, I'm just trying to make sure my understanding is correct.)

I guess this is fine, I'm just almost a little bummed it needs a condition in the template at all rather than just letting range over an empty list quietly DTRT.  Any time zero isn't a special case I'm happier.

Reply:
It did generate dead code, I'm 100% sure. I'm just not going to bother spending half an hour digging up what the original code looked like :P If vet complained about "no-op" switches as dead code, there would be a lot of angry people about it.

You seem to prefer simpler generation templates, as do I, so I'll merge this.